From 0b9f4431e9b7959044fac25f49bc92943737760f Mon Sep 17 00:00:00 2001
From: Sebastian Ott <sebott@redhat.com>
Date: Fri, 16 May 2025 13:01:06 +0200
Subject: [PATCH] hw/i386: Fix machine type compatibility

RH-Author: Sebastian Ott <sebott@redhat.com>
RH-MergeRequest: 453: hw/i386: Fix machine type compatibility
RH-Jira: RHEL-92430
RH-Acked-by: Cornelia Huck <cohuck@redhat.com>
RH-Acked-by: Miroslav Rezanina <mrezanin@redhat.com>
RH-Commit: [1/1] f41bb758603453ef6e03b57f144b998b83b5fe04

Upstream Status: RHEL only

Ensure compatibility of rhel specific pc-q35-rhel9.4.0 machine type.
Pick up missing bits from pc_compat_9_0 upstream.

Signed-off-by: Sebastian Ott <sebott@redhat.com>
---
 hw/i386/pc.c         | 10 ++++++++++
 hw/i386/pc_q35.c     |  2 ++
 include/hw/i386/pc.h |  3 +++
 3 files changed, 15 insertions(+)

diff --git a/hw/i386/pc.c b/hw/i386/pc.c
index 7af762065f..3447e44b97 100644
--- a/hw/i386/pc.c
+++ b/hw/i386/pc.c
@@ -298,6 +298,16 @@ GlobalProperty pc_rhel_compat[] = {
 };
 const size_t pc_rhel_compat_len = G_N_ELEMENTS(pc_rhel_compat);
 
+GlobalProperty pc_rhel_10_0_compat[] = {
+    /* pc_rhel_10_0_compat from pc_compat_9_0 */
+    { TYPE_X86_CPU, "x-amd-topoext-features-only", "false" },
+    { TYPE_X86_CPU, "x-l1-cache-per-thread", "false" },
+    { TYPE_X86_CPU, "guest-phys-bits", "0" },
+    { "sev-guest", "legacy-vm-type", "on" },
+    { TYPE_X86_CPU, "legacy-multi-node", "on" },
+};
+const size_t pc_rhel_10_0_compat_len = G_N_ELEMENTS(pc_rhel_10_0_compat);
+
 GlobalProperty pc_rhel_9_3_compat[] = {
     /* pc_rhel_9_3_compat from pc_compat_8_0 */
     { "virtio-mem", "unplugged-inaccessible", "auto" },
diff --git a/hw/i386/pc_q35.c b/hw/i386/pc_q35.c
index b2b8124225..4a326f2f99 100644
--- a/hw/i386/pc_q35.c
+++ b/hw/i386/pc_q35.c
@@ -714,6 +714,8 @@ static void pc_q35_rhel_machine_9_4_0_options(MachineClass *m)
 
     compat_props_add(m->compat_props, hw_compat_rhel_10_0,
                      hw_compat_rhel_10_0_len);
+    compat_props_add(m->compat_props, pc_rhel_10_0_compat,
+                     pc_rhel_10_0_compat_len);
     compat_props_add(m->compat_props, hw_compat_rhel_9_5,
 		     hw_compat_rhel_9_5_len);
 }
diff --git a/include/hw/i386/pc.h b/include/hw/i386/pc.h
index 61609027d0..2befb4453a 100644
--- a/include/hw/i386/pc.h
+++ b/include/hw/i386/pc.h
@@ -305,6 +305,9 @@ extern const size_t pc_compat_2_3_len;
 extern GlobalProperty pc_rhel_compat[];
 extern const size_t pc_rhel_compat_len;
 
+extern GlobalProperty pc_rhel_10_0_compat[];
+extern const size_t pc_rhel_10_0_compat_len;
+
 extern GlobalProperty pc_rhel_9_3_compat[];
 extern const size_t pc_rhel_9_3_compat_len;
 
-- 
2.39.3

